examples/glue_benchmark/glue.py

```diff
@@ -158,26 +175,39 @@ def generate_submission_files(finetuning_model, test_ds):
     if not os.path.exists(FLAGS.submission_directory):
         os.makedirs(FLAGS.submission_directory)
     filename = FLAGS.submission_directory + "/" + filenames[FLAGS.task_name]
-
     labelname = labelnames.get(FLAGS.task_name)
+
+    predictions = finetuning_model.predict(test_ds)
+    if FLAGS.task_name == "stsb":
+        predictions = np.squeeze(predictions)
+    else:
+        predictions = np.argmax(predictions, -1)
+
+    # Map the predictions to the right index order.
+    contents = ["" for _ in idx_order]
+    for i, pred in enumerate(predictions):
```

Comment:
if `idx_order` was instead a dataset `test_idx_ds` or something like that, you could just do `for idx, pred in zip(test_idx_ds.as_numpy_iterator(), predictions)`

And then below `contents[idx] = pred_value`

Reply:
even if this is still a list (re above comment) `for idx, pred in zip(idx_order, predictions)` would be a bit more readable here!

Reply:
Yea, will use this style, thanks!